### PR TITLE
Adapt for Hazelcast 4.0

### PR DESCRIPTION
**Pull request by leszko**
The main difference is that now Management Center connects to Hazelcast cluster instead of Hazelcast connecting to Management Center.

### CODE REVIEW

File: stable/hazelcast-enterprise/templates/mancenter-statefulset.yaml [hunk 1]
```diff
@@ -107,6 +107,8 @@ spec:
         {{- else }}
           value: {{ .Values.hazelcast.licenseKey }}
         {{- end }}
+        - name: MC_INIT_CMD
+          value: "./mc-conf.sh cluster add -H /data -cn {{ if index .Values "hazelcast" "yaml" "hazelcast" "cluster-name" }}{{ index .Values "hazelcast" "yaml" "hazelcast" "cluster-name" }}{{ else }}dev{{ end }} -ma {{ template "hazelcast.serviceName" . }}"
```
Comment (mesutcelik) [inline stable/hazelcast-enterprise/templates/mancenter-statefulset.yaml]: is this making `cluster-name` specified in values.yaml available in MC?

File: stable/hazelcast-enterprise/templates/statefulset.yaml [hunk 2]
```diff
@@ -124,7 +124,7 @@ spec:
...
+          value: "-Dhazelcast.config=/data/hazelcast/hazelcast.yaml -DserviceName={{ template "hazelcast.serviceName" . }} -Dnamespace={{ .Release.Namespace }} -Dhazelcast.ssl={{ .Values.hazelcast.ssl }} -Dhazelcast.hotRestart={{ .Values.hotRestart.enabled }} {{ if .Values.gracefulShutdown.enabled }}-Dhazelcast.shutdownhook.policy=GRACEFUL -Dhazelcast.shutdownhook.enabled=true -Dhazelcast.graceful.shutdown.max.wait={{ .Values.gracefulShutdown.maxWaitSeconds }} {{ end }} -Dhazelcast.cluster.version.auto.upgrade.enabled={{ .Values.hazelcast.updateClusterVersionAfterRollingUpgrade }} {{ if .Values.metrics.enabled }}-Dhazelcast.jmx=true{{ end }} {{ .Values.hazelcast.javaOpts }}"
```
Comment (mesutcelik) [inline stable/hazelcast-enterprise/templates/statefulset.yaml]: it is not really clear what has changed here?

File: stable/hazelcast-enterprise/templates/mancenter-statefulset.yaml [hunk 3]
```diff
@@ -62,7 +62,7 @@ spec:
         {{- if .Values.mancenter.livenessProbe.enabled }}
         livenessProbe:
           httpGet:
-            path: /hazelcast-mancenter/health
```
Comment (mesutcelik) [inline stable/hazelcast-enterprise/templates/mancenter-statefulset.yaml]: no more `hazelcast-mancenter` context path?

File: stable/hazelcast-enterprise/templates/statefulset.yaml (hunk 2 above)
Comment (hasancelik) [reply stable/hazelcast-enterprise/templates/statefulset.yaml]: It seems all parameters that contain `mancenter` keyword are removed.

File: stable/hazelcast-enterprise/templates/mancenter-statefulset.yaml (hunk 1 above)
Comment (leszko) [reply stable/hazelcast-enterprise/templates/mancenter-statefulset.yaml]: yes

File: stable/hazelcast-enterprise/templates/mancenter-statefulset.yaml (hunk 3 above)
Comment (leszko) [reply stable/hazelcast-enterprise/templates/mancenter-statefulset.yaml]: yeah, finally they changed it.

**Pull request merged**